# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: ipc/glue/CrashReporterHost.h
# Commit: bff57b2e4b16
# Full Hash: bff57b2e4b16fbcc4b5bf881c66a352ed7e2bd92
# Author: Gabriele Svelto <gsvelto@mozilla.com>
# Date: 2022-09-12 21:40:55
# Regressor Bug: 1789421
# File Overlap Count: 1
# Description:
#   Bug 1789421 - Generate crash reports for content process shutdown hangs only when we can confirm that the process did not shut down in time r=jstutte
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D156838
# ==============================================================================

diff -r f9922dd645d5 -r bff57b2e4b16 ipc/glue/CrashReporterHost.h
--- a/ipc/glue/CrashReporterHost.h	Mon Sep 12 08:56:33 2022 +0000
+++ b/ipc/glue/CrashReporterHost.h	Mon Sep 12 08:57:16 2022 +0000
@@ -46,7 +46,10 @@
   // If a minidump was already captured (e.g. via the hang reporter), this
   // finalizes the existing report by attaching metadata, writing out the
   // .extra file and notifying the crash service.
-  bool FinalizeCrashReport();
+  void FinalizeCrashReport();
+
+  // Delete any crash report we might have generated.
+  void DeleteCrashReport();
 
   // Generate a paired minidump. This does not take the crash report, as
   // GenerateCrashReport does. After this, FinalizeCrashReport may be called.